Updated old repo

Jassob opened the pull request.

Jassob committed:
Merge pull request #1 from Jassob/master

Updated DNollK.se from upstream

tairoman committed:
Bytte modul

Jassob committed:
Removed the about/img/ prefix to images.

Jassob committed:
Merge pull request #2 from Jassob/develop

Updated DNollK.se develop branch to upstream

Jassob committed:
Made publishedPosts a static method.

Jassob committed:
Merge branch 'develop' of github.com:dtekcth/DNollK.se into develop

Jassob committed:
Updated develop to match the current master branch.

Develop branch was quite out of date.

Jassob committed:
Made uploads filename more reasonably.

Uploaded images are now stored in
STATIC_ROOT/uploads/YYYY/uploadName.extension
instead of STATIC_ROOT/uploads/imageFileName.extension, where YYYY are
the year the images were uploaded.

This is only the case when there is no other file in that directory with
the same name and extension. In that case there will be some random
characters appended to the filename, which more or less destroys the
purpose of this commit.

Maybe we will get to implement the route functionality soon...

tairoman committed:
Added static files

tairoman committed:
Merge branch 'master' of github.com:dtekcth/DNollK.se

Jassob committed:
Updated header

Jassob committed:
Layout and specifically titles are more DRY.

Before we rewrote basically the whole title for every page, even though
all were of the format PAGENAME - DNollK YEAR, where YEAR was hardcoded
string like 2015.

This is error prone as we would only need to miss one page to make an
inconsistent and rather poor impression on the viewer.
Now, however, in the base layout I've defined the title as PAGE - DNollK
YEAR, where YEAR comes from a template tag returning the current year.
PAGE is a new block which means it can be shadowed by each template.

Jassob committed:
Trailing spaces.

Jassob committed:
Made news by default only retrieve news from the current year.

Jassob committed:
Documented some functions.

Jassob committed:
Fixed a typo.

A closing curly bracket was forgotten in about/dnollk.dtl page variable.

Jassob committed:
Sacrificed readability and gained readability.

Sacrificed some readability in templates to make the generated HTML more
structured and easier to read.

Jassob committed:
Refactored the logic and made it more modular.

Jassob committed:
Updated humans.txt

Jassob committed:
Post model now uses date instead of datetime.

There were no reason to use datetime over date and only added
complexity. That is now hopefully fixed.
Also added methods that are more flexible in retrieving the posts and
can get posts from any year.

Jassob committed:
Added urls and view methods for news.

Jassob committed:
Added the first tests in the project!

tairoman committed:
Fix for upload problem!

tairoman committed:
Update settings.py

tairoman committed:
Added optional fields and donk and sektionen now has the same functionality

tairoman committed:
Made nick optional

Jassob committed:
Merge pull request #3 from dtekcth/upload-fix

Mergar in upload-fix i feature/portability

Jassob committed:
Added some kind of todo.

Yes, it is ok to use the Issue tracker as well.

Jassob committed:
Fixes #4.

Updated the css file to make the dropdown menu vertical instead of horizontal.

Jassob committed:
Fixes #4.

Updated the css file to make the dropdown menu vertical instead of horizontal.

Jassob committed:
Attempts to fix issue #8.

The value was chosen after an "emperic study".

Jassob committed:
Attempts to fix issue #8.

The value was chosen after an "emperic study".

Jassob committed:
Revert "Attempts to fix issue #8."

This was not supposed to be pushed to master branch yet.

Jassob committed:
Added static files

Jassob committed:
Updated header

Jassob committed:
Added forgotten migrations...

Jassob committed:
Fixed nollenkäten.

Jassob committed:
Fixed nollenkäten.

Jassob committed:
Updated nolldeklaration url.

Jassob committed:
Merge branch 'develop' into feature/portability

Jassob committed:
Merge branch 'feature/portability' into develop

This resolves #9.

Jassob committed:
Shrunk the logo from 43M to 94K..

Jassob committed:
Added image_id field

Jassob committed:
Actually added the field in the model

Jassob committed:
Removed old image field

Jassob committed:
New logo from DNollK.

Jassob committed:
Fixed error where uploads are saved outside of base dir

Jassob committed:
DNollK is no longer DONK.

Fixed that latest DNollK (hopefully current) showed up in 'Om DONK'.

Jassob committed:
Removed obsolete attributes.

Jassob committed:
Removed stray div.

Jassob committed:
Updated contact.views for portability.

Jassob committed:
Added date field to uploads.

This is mostly to make the admin view filterable between years.

Jassob committed:
Updated the year for the nollmodul.

Jassob committed:
Added alt fields on uploads.

Jassob committed:
Merge branch 'feature/uploads_for_images' into feature/alt_msg_in_uploads

Jassob committed:
Added template tag for rendering uploaded images.

This is a step on solving #11.

Jassob committed:
Updated the about app to use render_image tag.

Related to issue #11.

Jassob committed:
Removed uploads are removed from fs.

When removing an uploaded photo now also deletes the actual file from
the file system.

Jassob committed:
Added a upload field to Posts.

To make it easier for us and for authors to create posts we included an
upload field that gets rendered if there is an image linked to the post.

Jassob committed:
Index on news app now renders included images.

Jassob committed:
Updated news/models.py to reflect the migration.

Jassob committed:
Made Post sort posts in chronological order.

Posts for the current year (Post.published_posts method) are sorted in
descending order, otherwise posts start with oldest first.

tairoman commented on upload/models.py:
```diff
@@ -50,6 +50,12 @@ def save(self, *args, **kwargs):
         self.photo.name = self.name + '.' + ext
         super(Upload, self).save(*args, **kwargs)
 
+    def url(self, *args, **kwargs):
```
Varför har vi en urlmetod här? Om du vill komma åt var bilden är sparad så tar du {{ obj.photo.url }} ? Eller har jag missförstått?

Jassob replied:
Ehmm.. för att jag inte läser på ordentligt på API-referenserna... :+1: 

@tairoman, vänta med att merga tills denna metod är borta och den inte används någon mer stans.

tairoman commented on about/views.py:
```diff
@@ -57,7 +57,7 @@ def donk(request):
     "about/donk.dtl template".
     """
-    committees = Committee.objects.filter(name__startswith="DNollK-").order_by("-name")
+    committees = Committee.objects.filter(name__startswith="DNollK-").order_by("-name")[1:]
```
Smart! Vi kan skriva en snabb kommentar bredvid. För någon som kollar snabbt på koden för templatetaggen så makear det ingen sense att 2016 inte dyker upp på donk när taggen tar allt som har ett "-" i sitt namn.

Jassob replied:
Absolut! Dokumentation är bra

tairoman commented on upload/templatetags/upload_tags.py:
```diff
@@ -0,0 +1,14 @@
+register = template.Library()
+
+@register.simple_tag
+def render_image(upload, **kwargs):
```
Detta visar hur koolt templatetags är!

Jassob replied:
*host* get_year *host host*

tairoman commented:
Helt briljant! Jag hade inte kommit i närheten av en lösning som denna. Kollar på merge imorgon bitti!

Jassob committed:
Fixed so upload.url returns the real url.

Before this commit the url method only prepended '/uploads/' which of
course is not correct if we use another MEDIA_URL.
Now we prepend MEDIA_URL instead which is much more sensible.

Jassob committed:
Removed the redundant url method.

Apparently the models.Filefield contains a url-property to the uploaded
file already.

Jassob commented on upload/models.py:
```diff
@@ -80,7 +74,7 @@ class and style of the rendered upload.
 
-        return '<img src="' + self.url() + '" alt="' + self.alt +'" ' + elemId + ' ' + elemClass + elemStyle + '/>'
+        return '<img src="' + self.photo.url + '" alt="' + self.alt +'" ' + elemId + ' ' + elemClass + elemStyle + '/>'
 
```
@tairoman, nu är upload.url() borta :+1: 

Eftersom vi använder oss av render() i typ alla fall borde vi inte behöva laga på fler platser än här.

tairoman replied:
Och det funkar fint?

Jassob replied:
ehm.. jag har inte dragit ned det på dnollk.se eller dev.dnollk.se än, men det funkade på min lokala instans.

Om du laddar ner de nya commitsen till dev.dnollk.se (test-mapppen) så kan jag lösa en reload av uwsgi snart!

emilhem committed:
Improved animation for sidebar

Now the navigation sidebar will be hidden from 0 - 549px instead of beginning to be visible at 450px. Same with the content.

The navigation timings are improved as well by using delay instead of animating for 1s on showing and 0.7s on hiding. Now the animation is always 0.7s but has a 0.1s delay when showing.

emilhem committed:
Improved toggleSidebar functionality

When the content is finished being pushed it should hide to prevent side scrolling on the page (maybe the performance is better as well). It should be made visible before being pushed back to make sure we display the fancy transition.

Perhaps the same thing should be done to the sidebar, hiding it when it is out of view.

This might solve issue #17

tairoman committed:
Merge pull request #15 from dtekcth/feature/uploads_for_images

Feature/uploads for images

Jassob committed:
Merge branch 'develop' into emilhem-patch-1

Resolves #19.

Jassob committed:
Merge branch 'emilhem-patch-1' into develop

This resolves #19.

Improved animation for sidebar.

tairoman committed:
Faster mobile sidebar

tairoman committed:
Is fix

Jassob committed:
Implemented pagination for news.

This commit is related to issue #14.

Apparently Django has inbuilt support for pagination, so the
implementation was much easier than I anticipated.
I would have liked to only retrieve the results that I was interested in
from the DB, but this was the official and documented way to do it.

Jassob committed:
Typo...

Also related to issue #14.

Jassob committed:
Updated README.md

Jassob committed:
Updated README.md

Jassob committed:
Added some kind of pagination images to news.

Jassob committed:
feature/pagination - Fixed a dislocated endif.

Jassob committed:
Fixed buttons with shadow.

Jassob committed:
Added page navigation buttons.

Jassob committed:
Updated the page navigation to work on mobile.

Jassob committed:
Made it possible to request past events.

Jassob committed:
Added a generic pagination index view function.

Also moved all pagination stuff to dnollkse instead of news.

Jassob committed:
Made events paginated.

It uses the paginated-layout-with-sidebar template.

Jassob committed:
Fixed a forgotten view.

Jassob committed:
Revert "Updated README.md"

It really has nothing to do with this branch..
This reverts commit 4dc9f8d35e7c8088f2934a22d18dcc6295851aa7.

Jassob committed:
Add filtering on event date on admin view.

tairoman committed:
Merge pull request #22 from dtekcth/feature/pagination

Feature/pagination

Jassob committed:
Updated the page links.

Jassob committed:
Changed order on posts.

Made all posts be retrieved in chronological descending order.

Jassob committed:
Updated page links on events.

Jassob committed:
Merge branch 'emilhem-patch-2' of github.com:dtekcth/DNollK.se into develop

Jassob committed:
More CSS and less javascript.

Attempt to make more transitions use css instead of javascript.

Jassob committed:
Added comments...

emilhem committed:
Updated alt tag for header image

Jassob committed:
Updated the Admin view for committees.

The admin view is actually viewable now on screens < 30".
Also fixes #21.

tairoman committed:
Merge pull request #24 from dtekcth/develop

Some kind of release

Jassob committed:
Attempt to fix strange ordering on Events.

Jassob committed:
Merge branch 'develop'

emilhem committed:
Move scripts to fix undefined variables error

Moving function.js below jQuery makes it not crash since function.js needs $.

Jassob committed:
Removed sidebar button.

There are views without sidebar markup, but the sidebar button was
always shown on the mobile version of the site. This resulted in a
button that did nothing, except hiding the content on the page.

Jassob committed:
Moved description to base template.

Jassob committed:
Merge branch 'develop'
